nntrainer/layers/layer_context.h

```diff
@@ -47,10 +47,13 @@ class InitLayerContext {
     input_dim(dim),
     in_place(in_place_),
     num_outputs(num_out),
-    name(n) {
+    name(n),
+    prefix("") {
     NNTR_THROW_IF(!validate(), std::invalid_argument)
       << "Invalid init context name: " << name
       << " num inputs: " << getNumInputs();
+    if (prefix.empty())
+      prefix = name; // default prefix is the name
```

Comment:
I think we need to give this info say the default name prefix is name to custom layer developer. else if they are confused a little bit.

Reply:
how about s/prefix/scope

and if scope is not given, we don't set scope?

Reply:
Yes, we need to make a document/guide on how to make a custom layer, and it must include this information.

Reply:
Some scope will have to be set in order to make the names unique.